src/NuGet.Clients/NuGet.CommandLine/Common/SelfUpdater.cs

```diff
@@ -115,6 +124,17 @@ internal async Task UpdateSelfFromVersionAsync(string exePath, bool prerelease,
                             throw new CommandException(LocalizedResourceManager.GetString("UpdateCommandUnableToLocateNuGetExe"));
                         }
 
+                        //Verify nuget.exe authenticode signature if nuget.org is the package source
+                        if (string.Equals(source.Source, NuGetConstants.V3FeedUrl, StringComparison.Ordinal))
```

Comment:
should we do a V2 check as well?

Reply:
FWIW, according to VS restore telemetry in the last 2 days, just under 2% of restores used nuget.org's V2 URL. 2/3rd used it by itself, and 1/3rd (so, about 0.6%) use both V3 and V2 urls 🤦‍♂️

Reply:
I noticed that there are 4 URL patterns where source can be configured as nuget.org.

https://github.com/NuGet/NuGet.Client/blob/b56eacca7b4d73463631ec82c4b52867c13482b2/src/NuGet.Core/NuGet.Configuration/PackageSource/NuGetConstants.cs#L11-L15

I found a method `NuGet.VisualStudio.Common/Telemetry/TelemetryUtility.cs` that does exactly what I was looking for. 

https://github.com/NuGet/NuGet.Client/blob/b56eacca7b4d73463631ec82c4b52867c13482b2/src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/TelemetryUtility.cs#L79-L82

Unfortunately, that method is in `NuGet.VisualStudio.Common` and hence can't access it from `NuGet.CommandLine` project. I don't like copying code but created a new utility method in `NuGet.CommandLine` project to check if the package source is nuget.org. I will create a follow-up PR to refactor `IsNuGetOrg` code for better maintainability.